src/transformState.js

```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const actionObj of actions) {
+    switch (actionObj.type) {
+      case 'addProperties' :
+        Object.assign(state, actionObj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of actionObj.keysToRemove) {
+          delete state[key];
+        }
+        break;
+
+      case 'clear':
+        for (const element in state) {
+          if (state.hasOwnProperty(element)) {
+            delete state[element];
+          }
```

Comment:
you sort out all keys of the object, they are in this object

```suggestion
          delete state[element];
```